### DISCUSSION TITLE
Is there a way to re-export enums that are defined in a different Rust crate?

### DISCUSSION THREAD

**Discussion by DanielT** [PyO3/pyo3#3368] [Questions] [answered]
I have a rust crate which defines several large enums full of constants.

Now I would like to create python bindings for my crate.
For all of the structs and their methods, I can just use wrapper structs, e.g. PyMyClass(MyClass).
This seems like it will not work for enums.

Is there a way to re-export existing enums to make them visible to Python?

**Comment by adamreichold** [comment, marked as answer]
The only option I can currently think that does not involve extra infrastructure (like writing proc macros and such) would be tighter integration, i.e. instead of having separate types which provide the Python bindings, have `pyo3` as an optional dependency in the original crate and do something like

```rust
#[cfg_attr(feature = "pyo3", pyclass)]
pub enum TheEnum {
```

(Note that this still has some syntax limitations which are being worked one, c.f. #2786.)

One completely different approach would be considering that Python is highly dynamic and produce enums and/or attributes on-the-fly by using something like [`strum`](https://docs.rs/strum) to iterate the enum variants at runtime.

**Comment by DanielT** [comment]
Thanks for the answer.

It looks like it will take some experimentation to find a good way of doing this.
